enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -634,6 +635,66 @@ func (np *nodePool) fetchExecutionNodes(ctx context.Context) ([]*executionNode,
 	return executors, nil
 }
 
+// weightedResample resamples the slice `nodes` by weighting each node by its
+// assignable CPU. This means that larger nodes have an opportunity to appear
+// more often than smaller nodes, increasing their chance of being hit.
+//
+// To ensure fairness and diversity (mostly for tests with small number of
+// nodes):
+//   - the returned number of nodes is (maxSize / minSize) times greater than
+//     the original number of nodes
+//   - all original nodes will be returned at least once
+func weightedResample(nodes []*executionNode) []*executionNode {
+	if len(nodes) == 0 {
+		return nodes
+	}
+	largest := float64(0)
```

Comment:
probably fine as-is but you could just use nodes[i].GetAssignableMilliCpu() to avoid any weird edge cases here